src/commands/inhouse.js

```diff
@@ -0,0 +1,266 @@
+										new Discord.RichEmbed({title: 'Current Players:'})
+											.addField(joinedNames.join(', '), false)));
+									}
+								}
+							});
+						} else {
+							message.channel.send('Please include your League of Legends summoner name like so: !f inhouse join <summoner name>');
+						}
+						break;
+            } 
+        } catch (err) {
+            logger.error(err);
+        }
+    }
+} }
+
+class Inhouse {
+    constructor(_type) {
+        this.players = [];
+        this.currentMatch = null;
+        this.type = _type;
+    }
+
+    addPlayerToInhouse(newPLayerDiscordName, newPlayerSummonerName) {
+        return new Promise((resolve, reject) =>{
+            new Player(newPLayerDiscordName, newPlayerSummonerName).then(response => {
+                var newPlayer = response;
+                if(newPlayer.summonerID === -1) {
+                    resolve(-1);
+                }             
+                this.players.push(newPlayer);
+                if(this.players.length === 10 && this.currentMatch === null) {
+                    this.CreateMatch();
+                    resolve(0);
+                }
+    
+                resolve(1);
+            });
+        });   
+    }
+
+    CreateMatch() {
+        this.currentMatch = new Match(this.players, this.type);
+    }
+}
+
+class Player {
+    constructor(playerDiscordName, playerSummonerName) {
+        return new Promise((resolve, reject) => {
+            this.discordName = playerDiscordName;
+            this.summonerName = playerSummonerName;
+            this.summonerID;
+            this.rankWeight;
+    
+            this.getSummonerId(this.summonerName).then(response => {
+                this.summonerID = response;
+                if(this.summonerID !== null) {
+                    this.getRankWeight(this.summonerID).then(response => {
+                        this.rankWeight = response;
+                        resolve(this);
+                    });
+                }
+            });
+        });
+
+    }
+
+    getSummonerId(name) {
+        return new Promise((resolve, reject) => {
+            var _summonerId;
+            let fullUrl = urlSummId + name + urlFinish;
+        
+            https.get(fullUrl, (resp) => {
+                let data = '';
+                resp.on('data', (chunk) => {
+                    data += chunk;
+                    });
+                resp.on('end', () => {
+                    try {
+                        var jsonResp = JSON.parse(data);
+                        _summonerId = jsonResp.id;
+                        resolve(_summonerId);
+                    } catch(e) {
+                        reject(e.message);
+                    }
+                    });
+            }).on("error", (err) => {
+                reject(`Got error: ${err.message}`);
+            });
+        });
+    }
+    
+    getRankWeight(summId) {
+        return new Promise((resolve, reject) => {
+            var _rankWeight = 0;
```

Comment:
Added a linting rule for `no-var`. Pull latest and run lint when you get a chance.